liquibase-service/src/main/resources/db/rdb_modern/routines/005-sp_nrt_investigation_postprocessing-003.sql

```diff
@@ -146,6 +159,196 @@ BEGIN
+                                                        'INV_FORM_HEPGEN','INV_FORM_MEA','INV_FORM_PER','INV_FORM_RUB')
+                );
+
+
+        IF (@COUNT_LEGACY_INV>0)
+            BEGIN
+
+
+                SELECT nio.public_health_case_uid, nio.observation_id, nio.branch_id, nio.branch_type_cd
+                INTO #temp_inv_obs
+                FROM #temp_inv_table t
+                         LEFT JOIN dbo.nrt_investigation_observation nio on nio.public_health_case_uid = t.case_uid
+                WHERE nio.branch_type_cd = 'InvFrmQ';
+
+                IF @debug = 'true' SELECT '#temp_inv_obs', * FROM #temp_inv_table;
+
+                /*Coded*/
+                SELECT
+                    tnio.public_health_case_uid
+                     ,tnio.observation_id
+                     ,tnio.branch_id
+                     ,tnio.branch_type_cd
+                     ,c.cd
+                     ,c.response
+                INTO #tmp_coded
+                FROM #temp_inv_obs tnio
+                         INNER JOIN dbo.v_getobscode c on tnio.branch_id = c.branch_id
+                WHERE c.cd IN 	('INV153',	/* Import country*/
+                                  'INV154', 	/* state*/
+                                  'INV156', 	/* county*/
+                                  'INV128',	/* HSPTLIZD_IND*/
+                                  'RUB162',  /* DIE_FRM_THIS_ILLNESS_IND */
+                                  'MEA078',  /* DIE_FRM_THIS_ILLNESS_IND */
+                                  'PRT103', 	/* DIE_FRM_THIS_ILLNESS_IND */
+                                  'INV145',	/* DIE_FRM_THIS_ILLNESS_IND in Generic */
+                                  'INV149',  /* FOOD_HANDLR_IND */
+                                  'INV178',  /* PATIENT_PREGNANT_IND */
+                                  'INV148');  /* DAYCARE_ASSOCIATION_IND */
+
+                IF @debug = 'true' SELECT '#tmp_coded', * FROM #tmp_coded;
+
+                /*Text*/
+                SELECT
+                    tnio.public_health_case_uid
+                     ,tnio.observation_id
+                     ,tnio.branch_id
+                     ,tnio.branch_type_cd
+                     ,t.cd
+                     ,t.response
+                INTO #tmp_txt
+                FROM #temp_inv_obs tnio
+                         INNER JOIN dbo.v_getobstxt t ON tnio.branch_id = t.branch_id
+                WHERE t.cd = 'INV155'; /* city */
+
+                IF @debug = 'true' SELECT '#tmp_txt', * FROM #tmp_txt;
+
+                /*Numeric*/
+                SELECT
+                    tnio.public_health_case_uid
+                     ,tnio.observation_id
+                     ,tnio.branch_id
+                     ,tnio.branch_type_cd
+                     ,n.cd
+                     ,n.response
+                INTO #tmp_num
+                FROM #temp_inv_obs tnio
+                         INNER JOIN dbo.v_getobsnum n ON tnio.branch_id = n.branch_id
```

Comment:
should we use nolock on the usage of the view definition or keep it defined within the view definition ?

Reply:
Both can exist as well 👍 I'll update it accordingly.